### PR TITLE
Merge tag 'v0.31.3' into feature/attachments

### PR DESCRIPTION
**Pull request by turbolent**
## Description

Update the attachments feature branch with the latest Cadence release (v0.31.3).

______

- [ ] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Review by dsainati1 — APPROVED** [review]

**Review by SupunS — APPROVED** [review]

File: types.go [hunk 1]
```diff
@@ -117,6 +135,10 @@ func (AnyResourceType) ID() string {
 	return "AnyResource"
 }
 
+func (t AnyResourceType) Equal(other Type) bool {
+	return t == other
+}
```
Comment (turbolent) [inline types.go:140]: Had to add this

File: types.go [hunk 2]
```diff
@@ -136,6 +158,10 @@ func (AnyResourceAttachmentType) ID() string {
 	return "AnyResourceAttachment"
 }
 
+func (t AnyResourceAttachmentType) Equal(other Type) bool {
+	return t == other
+}
```
Comment (turbolent) [inline types.go:163]: Had to add this

File: types.go [hunk 3]
```diff
@@ -1163,6 +1383,16 @@ func (t *AttachmentType) Base() Type {
...
 
+func (t *AttachmentType) Equal(other Type) bool {
+	otherType, ok := other.(*AttachmentType)
+	if !ok {
+		return false
+	}
+
+	return t.Location == otherType.Location &&
+		t.QualifiedIdentifier == otherType.QualifiedIdentifier
+}
```
Comment (turbolent) [inline types.go:1394]: Had to add this

File: types.go [hunk 4]
```diff
@@ -98,6 +112,10 @@ func (AnyStructAttachmentType) ID() string {
 	return "AnyStructAttachment"
 }
 
+func (t AnyStructAttachmentType) Equal(other Type) bool {
+	return t == other
+}
```
Comment (turbolent) [inline types.go:117]: Had to add this

**Pull request merged**